feat: Added Add, Archive and Delete functionality

mnath30 opened the pull request:
Added the following functionalities:
1. Create New Note
2. Archive Note
3. Delete Note
4. Unarchive note
5. Restore deleted note
6. Delete note

The new note can be created from the Notes page and the note can be archived and moved to trash from the Notes page.
The Archive page contains all the archived notes and the notes can be unarchived and moved to trash from the Archive page
The Trash page contains all the notes added to trash and the notes can be moved to the archive page, deleted permanently and restored from trash. The restored notes will be added back to the Notes page.

mnath30 committed:
feat: Added Add, Archive and Delete functionality

Added the following functionalities:
1. Create New Note
2. Archive Note
3. Delete Note
4. Unarchive note
5. Restore deleted note
6. Delete note

mnath30 committed:
fix: Removed unused component

mnath30 committed:
feat: Removed unused code

mnath30 committed:
fix: Fixed the CSS of the sidebar

mnath30 committed:
fix : Fixed the border of side bar

dev-enforced commented on src/Components/Modal/Modal.jsx:
```diff
@@ -0,0 +1,104 @@
+  const { setShowModal } = modalClose;
+  const [title, setTitle] = useState("");
+  const [content, setContent] = useState("");
+  const [taglist, setTagList] = useState([]);
```
Instead of having multiple states you can create an object called as noteData having title content and tagList as keys. And the initial empty values can be stored in a variable known as defaultNoteData. And then you can use useState or Reducer hooks to help you change the values on the respective events.

mnath30 replied:
Made the suggested changes

dev-enforced commented on src/helper/archiveNote.js:
```diff
@@ -0,0 +1,10 @@
+  localStorage.setItem("archivednotes", JSON.stringify(newArchiveList));
+  const newAllNotesList = allnoteslist.filter((note) => note._id !== item._id);
+  localStorage.setItem("allnotes", JSON.stringify(newAllNotesList));
+  console.log(newArchiveList, newAllNotesList);
```
Can remove the console.log !!

dev-enforced commented on src/helper/deleteNote.js:
```diff
@@ -0,0 +1,10 @@
+const deleteNote = (item, deletelist, allnotelist) => {
+  const newDeletedList = [{ ...item }, ...deletelist];
+  const newNoteList = allnotelist.filter((note) => note._id !== item._id);
+  console.log(newDeletedList, newNoteList);
```
Here also you can remove the console.log

dev-enforced commented on src/reducer/noteReduce.js:
```diff
@@ -0,0 +1,99 @@
+        deletednotes: [...newDeletedList],
+      };
+    case "ARCHIVE_FROM_ALL_NOTES":
+      console.log("Reducerfunc", state.archivednotes, state.allnotes);
```
Here also you can remove the logger message

dev-enforced commented on src/reducer/noteReduce.js:
```diff
@@ -0,0 +1,99 @@
+        state.archivednotes,
+        state.allnotes
+      );
+      console.log("Reducerfunc", newArchiveList, newAllNotesList);
```
Similarly here too

mnath30 replied:
Removed all the console logs

dev-enforced commented on src/Components/Modal/Modal.jsx:
```diff
@@ -0,0 +1,104 @@
+      className="input-modal-body"
+      onClick={() => setShowModal((showModal) => !showModal)}
+    >
+      <div className="modal input-modal" onClick={(e) => e.stopPropagation()}>
```
What does e.stopPropagation do? Does it stop the form submit process?

mnath30 replied:
e.stopPropagation is used here so that the state change of modal which is applied on the parent div (which makes the modal close) is not carried over to the modal content i.e. the modal should not close on clicking on it.

dev-enforced commented on src/index.js:
```diff
@@ -5,16 +5,19 @@ import App from "./App";
-        <App />
-      </BrowserRouter>
+      <NoteProvider>
+        <BrowserRouter>
```
Why are you keeping the BrowserRouter provider below the custom providers made from your end

dev-enforced commented on src/index.js:
```diff
@@ -5,16 +5,19 @@ import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
 import { ThemeProvider } from "./Context/themeContext";
+import { NoteProvider } from "./Context/noteContext";
```
Can import all the context providers and useContext hooks in a single index.js file of your context folder. Making it easy for you to do multiple imports from context folder in a single LOC!!

dev-enforced reviewed (approved):
The code structure looks really good !! Good use of Local Storage and implementing the functionality related to Archives and Trash as shown in your preview: https://deploy-preview-2--my-notes-maker.netlify.app/
Had some minor queries and suggestions which you can find in this review?

dev-enforced reviewed (approved):
Liked the way you have developed the logic of shifting from archives,trash and main notes section and local Storage. Good work on this @mnath30 !!

nerdyspook reviewed (approved):
Liked the way you used localstorage.

mnath30 committed:
refactor: Refactored the code

Refactored the code to include the following:
1. Exported all the context from a single index folder.
2. Used a single useState for all the note data
3. Removed the console logs
4. Moved the browser router above all the custom provider

mnath30 commented:
Incorporated all the suggested changes.
